src/marqo/tensor_search/models/settings_object.py

```diff
@@ -117,12 +126,14 @@
                             "properties": {
                                 NsFields.hnsw_ef_construction: {
                                     "type": "integer",
+                                    "minimum": 1,
```

Comment:
ef_construction of 500 would actually be rather reasonable. In fact, the default is 512. We'd have to figure out what upper bounds we'd want to set. I don't think this would be self-evident.